### PR TITLE
Check that sales and edi is unique where create, write or copy

### PR DESCRIPTION
**Pull request by nan-tic-dev**
#165158

### CODE REVIEW

File: sale_edi.py [hunk 1]
```diff
@@ -961,38 +985,65 @@ def write(cls, *args):
...
-                    edi_sale.state = 'done'
-                    edi_sale.save()
-                    if not edi_sale.sale:
-                        continue
-                    raise UserError(
-                        gettext('sale_edi_ediversa.msg_edi_sale_with_sale',
-                            edi_sale=edi_sale.number,
-                            sale=(edi_sale.sale.number
-                                or edi_sale.sale.reference
-                                or edi_sale.sale.id)))
```
Comment (bernatnan) [inline sale_edi.py:978]: Is needed to remove the message and transactions.

**Review by bernatnan — CHANGES_REQUESTED** [review]
LGTM

File: sale_edi.py (hunk 1 above)
Comment (raimonesteve) [reply sale_edi.py:978]: d4c1e2f26e29de7263a6482d5417d01d78594d73

**Review by bernatnan — APPROVED** [review]
LGTM

**Pull request merged**